qiskit/transpiler/passes/mapping/depth_bounded_mapper.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# Copyright 2019 Andrew M. Childs, Eddie Schoute, Cem M. Unsal
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The depth-optimizing bounded depth mapper"""
+
+import logging
+import random
+import sys
+from typing import Mapping, Set, FrozenSet, Tuple, Optional, Iterable
+
+from qiskit.dagcircuit import DAGCircuit, DAGNode
+from qiskit.transpiler.passes.mapping.depth_mapper import DepthMapper, Reg, ArchNode
+from qiskit.transpiler.passes.mapping.mapper import Mapper
+from qiskit.transpiler.passes.mapping.placement import Placement
+
+logger = logging.getLogger(__name__)
+
+
+class BoundedDepthMapper(DepthMapper[Reg, ArchNode]):
+    """A depth mapper that will place the cheapest gate and move the rest closer.
+
+    After placing the most expensive gate, we place an upper bound on the movement cost
+    of the remaining gates that is the twice that of the most expensive. The mapper will
+    then move the remaining gates only as close as that upper bound lets it."""
+
+    def map(self,
+            circuit: DAGCircuit,
+            current_mapping: Mapping[Reg, ArchNode]) -> Mapping[Reg, ArchNode]:
+        """
+        Provides a permutation that maps the circuit to the architecture.
+
+        If a chosen mapping has a cost increase associated to it,
+        then we try to perform the operation locally instead.
+
+        :param circuit: A circuit to execute
```

Comment:
We are using Google style for the docstrings.

Reply:
Do you know of an automated way to convert this? It's quite a hassle to rewrite all my documentation.

Reply:
Never have to convert them, so not sure. Maybe https://stackoverflow.com/questions/44955366/simple-way-to-convert-python-docstrings-from-restructured-text-to-google-style

Reply:
Thanks, I used that tool and some manual fixing to convert the docstrings.